lib/api.js

```diff
@@ -70,6 +70,106 @@ export async function getallProjectsWithSlug() {
   return data?.allProjects
 }
 
+export async function getAllProjectsByAuthor(id) {
+  const data = fetchAPI(`
+    query getProjectsByAuthor($id: ItemId) {
+      allProjects(filter: {author: {eq: $id}}) {
+        id
+        date
+        title
+        slug
+        coverImage {
+          responsiveImage(imgixParams: {fm: jpg, fit: crop, w: 2000, h: 1000 }) {
+            ...responsiveImageFragment
+          }
+        }
+        excerpt
+        author {
+          slug
+          picture {
+            url
+          }
+        }
+        tags {
+          name
+        }
+      }
+    }
+    ${responsiveImageFragment}
+    `,
+    {
+      variables: {
+        id
+      }
+    }
+  )
+  return data;
+}
+
+export async function getAllProjectsByTag(id) {
+  const data = fetchAPI(`
+    query getProjectsByTag($tagId: [ItemId]) {
+      allProjects(filter: {tags: {anyIn: $tagId}}) {
+        id
+        date
+        title
+        slug
+        coverImage {
+          responsiveImage(imgixParams: {fm: jpg, fit: crop, w: 2000, h: 1000 }) {
+            ...responsiveImageFragment
+          }
+        }
+        excerpt
+        author {
+          slug
+          name
+          picture {
+            url(imgixParams: {fm: jpg, fit: crop, w: 100, h: 100, sat: -100})
+          }
+        }
+        tags {
+          id
+          name
+          slug
+        }
+      }
+    }
+    ${responsiveImageFragment}
+    `,
+    {
+      variables: {
+        id
```

Comment:
the other way to address this would be to change this to

```
tagId: id
```

as it stands, it is the same as
```
id: id
```